fjord/l10n/tests/test_ext.py

```diff
@@ -0,0 +1,162 @@
+
+    def test_gettext_is_safe(self):
+        """html in gettext is safe"""
+        tmpl = '{{ _("<b>blue towers rule!</b>") }}'
+        assert render(tmpl, LANGUAGE='en') == '<b>blue towers rule!</b>'
+        assert render(tmpl, LANGUAGE='xx') == '<b>BLUE TOWERS RULE!</b>'
+
+    def test_gettext_notsafe(self):
+        """interpolated html is not safe"""
+        tmpl = '{{ _("<b>%(foo)s</b>", foo="<p>") }}'
+        assert render(tmpl, LANGUAGE='en') == '<b>&lt;p&gt;</b>'
+
+    def test_gettext_format_notsafe(self):
+        """format html is not safe"""
+        tmpl = '{{ _("<b>{0}</b>").format("<p>") }}'
+        assert render(tmpl, LANGUAGE='en') == '<b>&lt;p&gt;</b>'
+
+    def test_trans(self):
+        tmpl = '<div>{% trans %}tower rules!{% endtrans %}</div>'
+        assert render(tmpl, LANGUAGE='en') == '<div>tower rules!</div>'
+        assert render(tmpl, LANGUAGE='xx') == '<div>TOWER RULES!</div>'
+
+    def test_trans_whitespace(self):
+        tmpl = (
+            '<div>\n'
+            '  {% trans %}\n'
+            '    Tower\n'
+            '    rules!\n'
+            '  {% endtrans %}\n'
+            '</div>\n'
+        )
+        assert (
+            render(tmpl, LANGUAGE='en') ==
+            (
+                '<div>\n'
+                '  Tower rules!\n'
+                '</div>'
+            )
+        )
+        assert (
+            render(tmpl, LANGUAGE='xx') ==
+            (
+                '<div>\n'
+                '  TOWER RULES!\n'
+                '</div>'
+            )
+        )
+
+    def test_trans_plural(self):
+        tmpl = (
+            '<div>\n'
+            '  {% trans count %}\n'
+            '    {{ count }} tower rules!\n'
+            '  {% pluralize %}\n'
+            '    {{ count }} towers rule!\n'
+            '  {% endtrans %}\n'
+            '</div>'
+        )
+        assert (
+            render(tmpl, count=1, LANGUAGE='en') ==
+            (
+                '<div>\n'
+                '  1 tower rules!\n'
+                '</div>'
+            )
+        )
+        assert (
+            render(tmpl, count=2, LANGUAGE='en') ==
+            (
+                '<div>\n'
+                '  2 towers rule!\n'
+                '</div>'
+            )
+        )
+        assert (
+            render(tmpl, count=1, LANGUAGE='xx') ==
+            (
+                '<div>\n'
+                '  1 TOWER RULES!\n'
+                '</div>'
+            )
+        )
+        assert (
+            render(tmpl, count=2, LANGUAGE='xx') ==
+            (
+                '<div>\n'
+                '  2 TOWERS RULE!\n'
+                '</div>'
+            )
+        )
+
+    def test_trans_interpolation(self):
+        tmpl = (
+            '{% trans tag="<p>" %}\n'
+            '  this is a tag <b>{{ tag }}</b>\n'
+            '{% endtrans %}'
+        )
+        assert (
+            render(tmpl, LANGUAGE='en') ==
+            'this is a tag <b>&lt;p&gt;</b>'
+        )
```

Comment:
Redid the extension tests so they're way easier to read/maintain. Plus they have a bunch of interpolation tests covering escaping.